packages/blockchain-tools/src/interfaces/Web3Server.ts

```diff
@@ -29,7 +39,7 @@ export default class Web3Server extends InterfaceBaseClass {
   setChain(slug: ChainSlugs) {
     this.chain = chainConfig[slug]
     this.network = getNetworkForChain(slug)
-    this.web3 = new Web3(this.network.rpcUrls.main)
+    this.web3 = new Web3(this.network?.rpcUrls?.main)
```

Comment:
this.network should always be defined, or should error because the slug is wrong